newsroom/public/views.py

```diff
@@ -20,7 +20,9 @@ def get_public_dashboard_config():
         return app.cache.get(PUBLIC_DASHBOARD_CONFIG_CACHE_KEY)
 
     config = get_resource_service("ui_config").get_section_config("home")
-    app.cache.set(PUBLIC_DASHBOARD_CONFIG_CACHE_KEY, config, timeout=app.config.get("DASHBOARD_CACHE_TIMEOUT", 300))
+    app.cache.set(
+        PUBLIC_DASHBOARD_CONFIG_CACHE_KEY, config, timeout=app.config.get("PUBLIC_CONTENT_CACHE_TIMEOUT", 240)
```

Comment:
what I mean is to use slightly different timeout for each app.cache.set call, maybe we should add some util for that like:
```py
def cache_timeout(timeout: int, extra=60):
    return timeout + random.randrange(1, extra)

app.cache.set(..., timeout=cache_timeout(app.config.get("PUBLIC_CONTENT_CACHE_TIMEOUT", 300))
```

Reply:
My reason for keeping the cache keys together are
* Public dashboard cache will purge once every 4 minutes
* Internal dashboard item cache is every 5 minutes
* There are 3 cache keys used for public access
  * GET UI config
  * GET card config
  * GET card items
* Keeping the above resources in sync, no chance of different cache keys having out of sync data
* 2/3 of the caches are for very small units of data (UI config, card config), so additional cache save cost is not much

Reply:
ok makes sense